smart/model.py

```diff
@@ -2199,3 +2256,49 @@ def adjust_dt(self):
         #     dt_scale = min(dt_scale * 0.8, 0.8)
         dt_cur = float(self.dt) * dt_scale
         self.set_dt(dt_cur)
+
+    def load_timesteps_from_xdmf(self, xdmffile):
+        times = []
+        tree = ET.parse(xdmffile)
+        for elem in tree.iter():
+            if elem.tag == "Time":
+                times.append(float(elem.get("Value")))
+        return times
+
+    def load_vector(self, h5_file, tVec):
+        cur_level = d.get_log_level()
+        d.set_log_level(40)  # suppress warning about rank of h5 data
+        with d.HDF5File(self.parent_mesh.mpi_comm, h5_file, "r") as cur_file:
+            # Find index associated with the starting time
+            has_timestamp = np.flatnonzero(np.isclose(tVec, float(self.t)))
+            if len(has_timestamp) > 0:
+                vec_new = d.Vector()
+                idx1 = has_timestamp[0]
+                cur_file.read(vec_new, f"VisualisationVector/{idx1}", True)
+            elif self.t > tVec[-1]:  # then starting after final time in the xdmf file
+                logger.warning(
+                    f"File {str(h5_file)} ends before current time {self.t}"
+                    "Using final time point in file instead."
+                )
+                vec_new = d.Vector()
+                idx1 = len(tVec) - 1
+                cur_file.read(vec_new, f"VisualisationVector/{idx1}", True)
+            elif self.t < tVec[0]:  # then starting before initial time in xdmf file
+                logger.warning(
+                    f"File {str(h5_file)} starts after current time {self.t}"
+                    "Using initial time point in file instead."
+                )
+                vec_new = d.Vector()
+                idx1 = 0
+                cur_file.read(vec_new, f"VisualisationVector/{idx1}", True)
+            else:  # then in between two times in the xdmf file
+                vec1 = d.Vector()
+                vec2 = d.Vector()
+                idx1 = np.flatnonzero(tVec < float(self.t))[-1]
```

Comment:
```suggestion
                idx1 = np.flatnonzero(tVec < float(self.t))[-1]
```
Should we check that `tVec` is always sorted? Or is this something that will always happen?